pkg/network/ovn_kubernetes.go

```diff
@@ -212,6 +215,26 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 	return objs, nil
 }
 
+// returns the value of mode found in the openshift-ovn-kubernetes/gateway-mode-config configMap
+// if it exists, otherwise returns whatever the global OVN_GATEWAY_MODE is set to (shared)
+func GetGatewayMode(kubeClient client.Client) string {
+	cm := &corev1.ConfigMap{
+		TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "ConfigMap"},
+		ObjectMeta: metav1.ObjectMeta{Name: "gateway-mode-config"},
+	}
+	nsn := types.NamespacedName{Namespace: "openshift-ovn-kubernetes", Name: "gateway-mode-config"}
+	err := kubeClient.Get(context.TODO(), nsn, cm)
+	if err != nil {
+		klog.V(2).Infof("Did not find gateway-mode-config. Using default mode: %s", OVN_GATEWAY_MODE)
+	}
+	if cm.Data["mode"] != "shared" && cm.Data["mode"] != "local" {
+		klog.V(2).Infof("Ignoring gateway-mode-config %s. Does not match \"shared\" or \"local\"", cm.Data["mode"])
```

Comment:
nit: Should this be a `klog.Warningf` rather than just a log?

Reply:
done